stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/service/sync/PushMessageSyncHandler.kt

```diff
@@ -73,31 +79,31 @@ public class PushMessageSyncHandler(private val service: Service) {
             return
         }
         createSyncNotificationChannel()
-        showForegroundNotification(notificationConfig.smallIcon)
+        startForegroundExecution(notificationConfig.smallIcon)
 
-        GlobalScope.launch(DispatcherProvider.IO) {
-            val cid: String = firebaseMessageParser.parse(message).let { "${it.channelType}:${it.channelId}" }
-            if (ChatDomain.isInitialized && ChatClient.isInitialized) {
-                logger.logD("Starting the sync")
-                performSync(ChatDomain.instance(), cid, ChatClient.instance(), message)
-            } else {
-                val syncConfig = syncModule.encryptedBackgroundSyncConfigStore.get()
-                syncConfig?.let {
-                    val config = it
-                    val user = User(id = config.userId)
-                    val token = config.userToken
-                    val client = initClient(service, user, token, config.apiKey)
-                    val domain = initDomain(user, client)
-                    logger.logD("Starting the sync, config: $syncConfig")
-                    performSync(domain, cid, client, message)
-                }
+        val cid: String = firebaseMessageParser.parse(message).let { "${it.channelType}:${it.channelId}" }
+        if (ChatDomain.isInitialized && ChatClient.isInitialized) {
+            logger.logD("Starting the sync")
+            performSync(ChatDomain.instance(), cid, ChatClient.instance(), message)
+        } else {
+            val syncConfig = syncModule.encryptedBackgroundSyncConfigStore.get()
+            syncConfig?.let {
+                val config = it
+                val user = User(id = config.userId)
+                val token = config.userToken
+                val client = initClient(service, user, token, config.apiKey)
+                val domain = initDomain(user, client)
+
+                logger.logD("Starting the sync, config: $syncConfig")
+
+                performSync(domain, cid, client, message)
             }
+            service.stopForeground(true)
```

Comment:
This only happens on the `else` branch now, seems suspicious

Reply:
Yes, my bad. Fixed.